### PR TITLE
Faster lora container and other nits

### PR DESCRIPTION
**Pull request by sordonia**
- main change in this PR is how experts are stored in LoRAContainer
- also rename CoalescedLoRAContainer to SkilledLoRAContainer (just for skilled loras)
- also, critical change to speed-up expert loading was to put an if to tie_params, tie_params routine seems to be very slow

### CODE REVIEW

File: mttl/models/expert_model.py [hunk 1]
```diff
@@ -21,10 +26,28 @@
...
+    """Context manager to disable all adapters in a model.
+
+    Args:
+        model (BaseExpertModel): The model to disable adapters in.
+    """
+    model.disable_adapters()
+
+    yield
+
+    model.enable_adapters()
```
Comment (pclucas14) [inline mttl/models/expert_model.py:48]: we should revert to the original state of the model.

File: mttl/models/library/expert.py [hunk 2]
```diff
@@ -262,7 +262,13 @@ def load_expert_from_hf_checkpoint(
...
     mttl_args_file = os.path.join(expert_path, MTTL_ARGS_NAME)
     if os.path.isfile(mttl_args_file):
-        training_config = torch.load(mttl_args_file, weights_only=False)
+        from mttl.arguments import AutoArgs
+
+        try:
+            training_config = torch.load(mttl_args_file, weights_only=False)
+            training_config = AutoArgs.fromdict(training_config)
+        except:
+            training_config = None
```
Comment (pclucas14) [inline mttl/models/library/expert.py:271]: can you add a comment saying when this fails  ? (If there is a situation in mind)

**Review by pclucas14 — APPROVED** [review]

File: mttl/models/containers/selectors/base.py [hunk 3]
```diff
@@ -534,3 +534,30 @@ def get_merging_weights(self, **selector_kwargs) -> Dict:
         raise NotImplementedError(
             "Not required for TaskNameSelector as it performs hard selection. Use 'get_expert_instance' instead."
         )
+
+
+@dataclass
+class KnowledgeExtractorSelectorConfig(SelectorConfig):
```
[the remark this answers is not in the document]
Comment (sordonia) [reply mttl/models/containers/selectors/base.py:540]: ok

File: mttl/models/expert_model.py (hunk 1 above)
Comment (sordonia) [reply mttl/models/expert_model.py:48]: what do you mean?

File: mttl/models/library/expert.py (hunk 2 above)
Comment (sordonia) [reply mttl/models/library/expert.py:271]: actually this will be removed

**Pull request merged**